components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -429,13 +443,62 @@ void BraveWalletService::SetDefaultWallet(mojom::DefaultWallet default_wallet) {
   }
 }
 
+void BraveWalletService::GetDefaultBaseCurrency(
+    GetDefaultBaseCurrencyCallback callback) {
+  std::move(callback).Run(::brave_wallet::GetDefaultBaseCurrency(prefs_));
+}
+
+void BraveWalletService::SetDefaultBaseCurrency(const std::string& currency) {
+  auto old_default_currency = ::brave_wallet::GetDefaultBaseCurrency(prefs_);
+  if (old_default_currency != currency) {
+    ::brave_wallet::SetDefaultBaseCurrency(prefs_, currency);
+  }
+}
+
+void BraveWalletService::GetDefaultBaseCryptocurrency(
+    GetDefaultBaseCryptocurrencyCallback callback) {
+  std::move(callback).Run(::brave_wallet::GetDefaultBaseCryptocurrency(prefs_));
+}
+
+void BraveWalletService::SetDefaultBaseCryptocurrency(
+    const std::string& cryptocurrency) {
+  auto old_default_cryptocurrency =
+      ::brave_wallet::GetDefaultBaseCryptocurrency(prefs_);
+  if (old_default_cryptocurrency != cryptocurrency) {
+    ::brave_wallet::SetDefaultBaseCryptocurrency(prefs_, cryptocurrency);
+  }
+}
+
 void BraveWalletService::OnDefaultWalletChanged() {
   auto default_wallet = ::brave_wallet::GetDefaultWallet(prefs_);
   for (const auto& observer : observers_) {
     observer->OnDefaultWalletChanged(default_wallet);
   }
 }
 
+void BraveWalletService::OnDefaultBaseCurrencyChanged() {
+  auto value = ::brave_wallet::GetDefaultBaseCurrency(prefs_);
+  for (const auto& observer : observers_) {
+    observer->OnDefaultBaseCurrencyChanged(value);
+  }
+}
+
+void BraveWalletService::OnDefaultBaseCryptocurrencyChanged() {
+  auto value = ::brave_wallet::GetDefaultBaseCryptocurrency(prefs_);
+  for (const auto& observer : observers_) {
+    observer->OnDefaultBaseCryptocurrencyChanged(value);
+  }
+}
+
+void BraveWalletService::OnNetworkListChanged() {
+  std::vector<mojom::EthereumChainPtr> chains;
+  brave_wallet::GetAllChains(prefs_, &chains);
+
+  for (const auto& observer : observers_) {
+    observer->OnNetworkListChanged(std::move(chains));
```

Comment:
This will be a use after move if there's more than one observer.  You could possibly just make the client query for the chains and not pass anything, or clone the vector list.

Reply:
Fixed in https://github.com/brave/brave-core/pull/10893/commits/0b07bedf76e8a6870ab8ce3536305f7dcdd78f91, thanks!